server/src/test/java/org/jetbrains/bsp/bazel/server/sync/languages/LanguagePluginServiceTest.kt

```diff
@@ -0,0 +1,382 @@
+
+class LanguagePluginServiceTest {
+
+    private var languagePluginsService: LanguagePluginsService? = null
+
+    @BeforeEach
+    fun beforeEach() {
+        val bazelInfo = BasicBazelInfo("", Paths.get(""))
+        val bazelPathsResolver = BazelPathsResolver(bazelInfo)
+        val jdkResolver = JdkResolver(bazelPathsResolver, JdkVersionResolver())
+        val javaLanguagePlugin = JavaLanguagePlugin(bazelPathsResolver, jdkResolver, bazelInfo)
+        val scalaLanguagePlugin = ScalaLanguagePlugin(javaLanguagePlugin, bazelPathsResolver)
+        val cppLanguagePlugin = CppLanguagePlugin()
+        val thriftLanguagePlugin = ThriftLanguagePlugin(bazelPathsResolver)
+        languagePluginsService = LanguagePluginsService(
+            scalaLanguagePlugin, javaLanguagePlugin, cppLanguagePlugin, thriftLanguagePlugin
+        )
+    }
+
+    @Nested
+    @DisplayName("Tests for the method shouldGetPlugin")
+    inner class ShouldGetPluginTest {
+
+        @Test
+        fun `should return JavaLanguagePlugin for Java Language`() {
+            // given
+            val languages: Set<Language> = HashSet.of(Language.JAVA)
+
+            // when
+            val plugin = languagePluginsService?.getPlugin(languages) as? JavaLanguagePlugin
+
+            // then
+            plugin shouldNotBe null
+        }
+
+        @Test
+        fun `should return JavaLanguagePlugin for Kotlin Language`() {
+            // given
+            val languages: Set<Language> = HashSet.of(Language.KOTLIN)
+
+            // when
+            val plugin = languagePluginsService?.getPlugin(languages) as? JavaLanguagePlugin
+
+            // then
+            plugin shouldNotBe null
+        }
+
+        @Test
+        fun `should return ScalaLanguagePlugin for Scala Language`() {
+            // given
+            val languages: Set<Language> = HashSet.of(Language.SCALA)
+
+            // when
+            val plugin = languagePluginsService?.getPlugin(languages) as? ScalaLanguagePlugin
+
+            // then
+            plugin shouldNotBe null
+        }
+
+        @Test
+        fun `should return EmptyLanguagePlugin for no Language`() {
+            // given
+            val languages: Set<Language> = HashSet.of()
+
+            // when
+            val plugin = languagePluginsService?.getPlugin(languages) as? EmptyLanguagePlugin
+
+            // then
+            plugin shouldNotBe null
+        }
+
+        @Test
+        fun `should return ThriftLanguagePlugin for Thrift Language`() {
+            // given
+            val languages: Set<Language> = HashSet.of(Language.THRIFT)
+
+            // when
+            val plugin = languagePluginsService?.getPlugin(languages) as? ThriftLanguagePlugin
+
+            // then
+            plugin shouldNotBe null
+        }
+    }
+
+
+    @Nested
+    @DisplayName("Tests for the method shouldGetSourceSet")
+    inner class ShouldGetSourceSetTest {
+        private var tmpRepo: Path? = null
```

Comment:
i think that u can `lateinit var` here as well, and init it in`@BeforeEach` and then use it in the following method

Reply:
Added lateinit and initialize it in @BeforeEach